lib/ContractInteract/GatewayComposer.js

```diff
@@ -0,0 +1,447 @@
+        arguments: args,
+      },
+    );
+  }
+
+  /**
+   * Method for request stake. This transfers value tokens from caller to
+   * GatewayComposer.
+   *
+   * @param {string} stakeVT Stake amount in wei.
+   * @param {string} mintBT Minted amount in wei.
+   * @param {string }gateway Address of gateway contract.
+   * @param {string} beneficiary Beneficiary address on auxiliary chain.
+   * @param {string} gasPrice Gas price that staker is willing to pay for the reward.
+   * @param {string} gasLimit Maximum gas limit for reward calculation.
+   * @param {string} nonce Staker nonce managed by gateway contract.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @returns {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async requestStake(
+    stakeVT,
+    mintBT,
+    gateway,
+    beneficiary,
+    gasPrice,
+    gasLimit,
+    nonce,
+    txOptions,
+  ) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address ${txOptions.from} in transaction options.`);
+      return Promise.reject(err);
+    }
+
+    const tx = await this.requestStakeRawTx(
+      stakeVT,
+      mintBT,
+      gateway,
+      beneficiary,
+      gasPrice,
+      gasLimit,
+      nonce,
+    );
+
+    return Utils.sendTransaction(tx, txOptions);
+  }
+
+
+  /**
+   * Raw transaction for request stake.
+   *
+   * @param {string} stakeVT Stake amount in wei.
+   * @param {string} mintBT Minted amount in wei.
+   * @param {string }gateway Address of gateway contract
+   * @param {string} beneficiary Beneficiary address on auxiliary chain.
+   * @param {string} gasPrice Gas price that staker is willing to pay for the reward.
+   * @param {string} gasLimit Maximum gas limit for reward calculation.
+   * @param {string} nonce Staker nonce managed by gateway contract.
+   *
+   * @returns {Promise<Object>} Promise that resolves to raw transaction.
+   */
+  requestStakeRawTx(
+    stakeVT,
+    mintBT,
+    gateway,
+    beneficiary,
+    gasPrice,
+    gasLimit,
+    nonce,
+  ) {
+    if (!new BN(stakeVT).gtn(0)) {
+      const err = new TypeError(
+        `Stake amount must be greater than zero: ${stakeVT}.`,
+      );
+      return Promise.reject(err);
+    }
+    if (!new BN(mintBT).gtn(0)) {
+      const err = new TypeError(
+        `Mint amount must be greater than zero: ${mintBT}.`,
+      );
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(gateway)) {
+      const err = new TypeError(
+        `Invalid gateway address: ${gateway}.`,
+      );
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(beneficiary)) {
+      const err = new TypeError(
+        `Invalid beneficiary address: ${beneficiary}.`,
+      );
+      return Promise.reject(err);
+    }
+    if (gasPrice === undefined) {
```

Comment:
In these cases the error could be that it is `missing` and you don't need to print `undefined` in that case.